internal/pkg/utils/testproject/project.go

```diff
@@ -223,20 +231,21 @@ func (p *Project) Clean() error {
 	if err != nil {
 		return errors.Errorf(`cannot fetch default branch in project "%d": %w`, p.ID(), err)
 	}
+
+	p.stateFilePath = ""
 	p.defaultBranch = defaultBranch
+	p.branchesByID = make(map[storageapi.BranchID]*storageapi.Branch)
+	p.branchesByName = make(map[string]*storageapi.Branch)
 
 	p.logf("■ Cleanup done.")
 	return nil
 }
 
 func (p *Project) SetState(stateFilePath string) error {
-	// Remove all objects
-	err := p.Clean()
-	if err != nil {
-		return err
+	if p.stateFilePath != "" {
+		return errors.New("SetState method can be called only once after the Clean method")
```

Comment:
Honza, kontroloval som o mazanie `buckets` v testoch ako si hlasil.

Jediny problem, ktory som nasiel je, ze sa `Clean` volalo az v `SetState`.

Teda tento unit test mohol mat duplicitne buckets, kedze nevola `SetState`.
https://github.com/keboola/keboola-as-code/blob/f7016c9a78f1a8b676025fef97a8a10e9121f119/internal/pkg/service/buffer/api/service/token_test.go#L15-L22

Povodne za tym bola taka logika, ze niekedy potrebujes testovaci projekt, 
ale netrapi ta co v nom je, napr. chces iba otestovat vytvorenie noveho tokenu.

Take use-cases mame v projekte mozno 3 a rovnako dobre tam posluzi cisty projekt.

A je to teda lepsie presunut do `init`, ze vzdy dostanes cisty projekt.
Predide sa tak nedorozumeniu ohladom toho, ako to funguje.

Reply:
Ked nad tym rozmyslam, tak aj v E2E testoch pre Buffer API sa nenastavuje ziadny stav projektu, takze to bolo aj tam :thinking:

Teda mas pravdu, ze to neslo.l